commands/kafka/cluster/list.go

```diff
@@ -0,0 +1,104 @@
+package cluster
+
+import (
+	"context"
+	"fmt"
+	"github.com/ionos-cloud/ionosctl/v6/commands/kafka/completer"
+	"github.com/ionos-cloud/ionosctl/v6/internal/client"
+	"github.com/ionos-cloud/ionosctl/v6/internal/constants"
+	"github.com/ionos-cloud/ionosctl/v6/internal/core"
+	"github.com/ionos-cloud/ionosctl/v6/internal/printer/json2table"
+	"github.com/ionos-cloud/ionosctl/v6/internal/printer/json2table/jsonpaths"
+	"github.com/ionos-cloud/ionosctl/v6/internal/printer/jsontabwriter"
+	"github.com/ionos-cloud/ionosctl/v6/internal/printer/tabheaders"
+	kafka "github.com/ionos-cloud/sdk-go-kafka"
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+)
+
+func List() *core.Command {
+	cmd := core.NewCommand(
+		context.Background(), nil, core.CommandBuilder{
+			Namespace: "kafka",
+			Resource:  "cluster",
+			Verb:      "list",
+			Aliases:   []string{"ls"},
+			ShortDesc: "Retrieve all clusters using pagination and optional filters",
+			Example:   `ionosctl kafka c list --location de/txl`,
+			PreCmdRun: func(c *core.PreCommandConfig) error {
+				if err := core.CheckRequiredFlags(c.Command, c.NS, constants.FlagLocation); err != nil {
+					return err
+				}
+
+				return nil
+			},
+			CmdRun: func(c *core.CommandConfig) error {
+				return listClusters(c)
+			},
+			InitClient: true,
+		},
+	)
+
+	cmd.AddStringFlag(constants.FlagFilterName, "", "", "Filter used to fetch only the records that contain specified name.")
+	cmd.AddSetFlag(constants.FlagFilterState, "", "", []string{"AVAILABLE", "BUSY", "DEPLOYING", "UPDATING", "FAILED_UPDATING", "FAILED", "DESTROYING"}, "Filter used to fetch only the records that contain specified state.")
+	_ = cmd.Command.RegisterFlagCompletionFunc(constants.FlagFilterState, func(c *cobra.Command, args []string, toComplete string) ([]string, cobra.ShellCompDirective) {
+		return []string{"AVAILABLE", "BUSY", "DEPLOYING", "UPDATING", "FAILED_UPDATING", "FAILED", "DESTROYING"}, cobra.ShellCompDirectiveNoFileComp
+	})
+
+	cmd.AddStringFlag(constants.FlagLocation, "", "", "The datacenter location", core.RequiredFlagOption())
```

Comment:
If I want to list clusters across all locations, I have to make multiple `list`s with multiple locations, right?